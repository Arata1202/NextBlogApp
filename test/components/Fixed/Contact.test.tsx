import { render } from '@testing-library/react';
import ContactPage from '@/components/Fixed/Contact';

const mockSidebarArticles = {
  contents: [
    { id: '1', title: '記事1', description: '概要1', url: '/articles/1' },
    { id: '2', title: '記事2', description: '概要2', url: '/articles/2' },
    { id: '3', title: '記事3', description: '概要3', url: '/articles/3' },
  ],
};

describe('ContactPage', () => {
  test('スナップショット（ContactPage）', () => {
    const { asFragment } = render(<ContactPage sidebarArticles={mockSidebarArticles} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
