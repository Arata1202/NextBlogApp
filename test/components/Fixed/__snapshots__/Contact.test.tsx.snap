// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactPage スナップショット（ContactPage） 1`] = `
<DocumentFragment>
  <div
    class="max-w-[85rem] sm:px-6 lg:px-8 mx-auto"
  >
    <div
      class="grid lg:grid-cols-3 gap-y-8 lg:gap-y-0 lg:gap-x-6"
    >
      <div
        class="lg:col-span-2"
      >
        <div
          class="FirstAd"
        />
        <div
          class=""
        >
          <div
            class="space-y-5 lg:space-y-8"
          >
            <div
              class="includeBanner flex justify-end gap-x-5"
            />
          </div>
          <p
            class="mt-5"
          >
            当ブログに関するご質問やお気づきの点がございましたら、お気軽にお問い合わせください。 お問い合わせから2～3日中にはご返信させていただきます。
          </p>
          <form
            class="pt-5 mb-5"
            method="POST"
          >
            <div
              class="grid grid-cols-1 gap-x-8 gap-y-6 sm:grid-cols-2"
            >
              <div
                class="sm:col-span-2"
              >
                <label
                  class="block text-sm font-semibold leading-6 LightTheme placeholder:text-gray-500"
                  for="first-name"
                >
                  氏名
                </label>
                <div
                  class="mt-2.5"
                >
                  <input
                    autocomplete="given-name"
                    class="hover:border-blue-500 block w-full rounded-md border py-2 pl-3 pr-3 sm:text-sm sm:leading-6 LightTheme"
                    id="sei"
                    name="sei"
                    type="text"
                  />
                </div>
              </div>
              <div
                class="sm:col-span-2"
              >
                <label
                  class="block text-sm font-semibold leading-6 LightTheme placeholder:text-gray-500"
                  for="last-name"
                >
                  題名
                </label>
                <div
                  class="mt-2.5"
                >
                  <input
                    autocomplete="family-name"
                    class="hover:border-blue-500 block w-full rounded-md border py-2 pl-3 pr-3 sm:text-sm sm:leading-6 LightTheme"
                    id="mei"
                    name="mei"
                    type="text"
                  />
                </div>
              </div>
              <div
                class="sm:col-span-2"
              >
                <label
                  class="block text-sm font-semibold leading-6 LightTheme placeholder:text-gray-500"
                  for="email"
                >
                  メールアドレス
                </label>
                <div
                  class="mt-2.5"
                >
                  <input
                    autocomplete="email"
                    class="hover:border-blue-500 block w-full rounded-md border py-2 pl-3 pr-3 sm:text-sm sm:leading-6 LightTheme"
                    id="email"
                    name="email"
                    type="text"
                  />
                </div>
              </div>
              <div
                class="sm:col-span-2"
              >
                <label
                  class="block text-sm font-semibold leading-6 LightTheme placeholder:text-gray-500"
                  for="message"
                >
                  内容
                </label>
                <div
                  class="mt-2.5"
                >
                  <textarea
                    class="hover:border-blue-500 block w-full rounded-md border py-2 pl-3 pr-3 sm:text-sm sm:leading-6 LightTheme"
                    id="message"
                    name="message"
                    rows="4"
                  />
                </div>
              </div>
            </div>
            <div
              class="mt-3"
            />
            <div
              class="mt-3"
            >
              <button
                class="block w-full rounded-md px-3.5 py-2.5 text-center text-sm font-semibold shadow-s border hover:border-blue-500 hover:text-blue-500 LightTheme"
                disabled=""
                type="submit"
              >
                送信
              </button>
            </div>
          </form>
          <div
            class="FirstAd"
          />
        </div>
      </div>
      <div
        class="mobile"
      />
    </div>
  </div>
  <div
    aria-live="assertive"
    class="confirmAlert pointer-events-none fixed inset-0 flex items-end px-4 py-6 sm:items-start sm:p-6"
  >
    <div
      class="flex w-full flex-col items-center space-y-4 sm:items-end"
    />
  </div>
</DocumentFragment>
`;
